crates/chain/src/keychain/txout_index.rs

```diff
@@ -207,19 +210,21 @@ const DEFAULT_LOOKAHEAD: u32 = 25;
 #[derive(Clone, Debug)]
 pub struct KeychainTxOutIndex<K> {
     inner: SpkTxOutIndex<(DescriptorId, u32)>,
-    // keychain -> (descriptor, descriptor id) map
-    keychains_to_descriptors: BTreeMap<K, (DescriptorId, Descriptor<DescriptorPublicKey>)>,
-    // descriptor id -> keychain set
-    // Because different keychains can have the same descriptor, we rank keychains by `Ord` so that
-    // that the first keychain variant (according to `Ord`) has the highest rank. When associated
-    // data (such as spks, outpoints) are returned with a keychain, we return the highest-ranked
-    // keychain with it.
-    descriptor_ids_to_keychain_set: HashMap<DescriptorId, BTreeSet<K>>,
+    // keychain -> descriptor_id map
+    keychains_to_descriptor_ids: BTreeMap<K, DescriptorId>,
+    // descriptor_id -> keychain set
+    // This is a reverse map of `keychains_to_descriptors`. Although there is only one descriptor
+    // per keychain, different keychains can refer to the same descriptor, therefore we have a set
+    // of keychains per descriptor. When associated data (such as spks, outpoints) are returned with
+    // a keychain, we return it with the highest-ranked keychain with it. We rank keychains by
+    // `Ord`, therefore the keychain set is a `BTreeSet`. The earliest keychain variant (according
+    // to `Ord`) has precedence.
+    keychains: HashMap<DescriptorId, BTreeSet<K>>,
```

Comment:
I have a second thought about this renaming. What do you think about: `ids_by_keychain` and `keychains_by_id`?

Reply:
My initial thought is, if we're simplifying `descriptor_ids_to_keychains` to something like `keychains_by_id`, would it make sense to do something similar to `keychains_to_descriptor_ids`?

Reply:
I meant `keychains_by_ids` to replace `descriptor_ids_to_keychains`. I was suggesting it because the wording makes more sense to me. However, `ids_to_keychains` can work too.